src/components/Question/Question.js

```diff
@@ -53,9 +53,34 @@ class Question extends Component {
     /* check if current question was already answered correctly */
     const alreadyAnswered =
       score.books[bookId].chapters[chapterId].questions[questionIndex].answered;
+    const alreadyCorrect =
+      score.books[bookId].chapters[chapterId].questions[questionIndex].correct;
 
-    let newScore = score;
-    const newBookScore = score.books.map((book, index) => {
+    /* calculate the numeric score difference 
+     * based on current score state */
+    const scoreDiff = {
+      correct: isCorrect && !alreadyCorrect ? 1 : 0,
+      incorrect: alreadyCorrect
+        ? 0 // previously answered correctly
+        : alreadyAnswered
+          ? isCorrect
+            ? -1 // previously answered, correct answer given
+            : 0 // previously answered, incorrect answer given
+          : isCorrect
+            ? 0 // first attempt, correct answer given
+            : 1, // first attempt, incorrect answer given
+    };
+
+    // return a new score object
+    // NOTE: possible bug; if we declare `newScore` by initializing to
+    // `score`'s properties ( newScore = { books: ..., } etc. )
+    // vs. initializing it as it is below (line 80)
+    // the result is that the newScore object doesn't actually
```

Comment:
Also, when implementing the updated score object, I ran into this weird bug(?). My goal was to remove `newScore` altogether and just contain the entire "newScore" object as an argument to `updateScore()` (line 92) without an additional variable declaration. However, this (lines 75 - 79) happens. 

The app behaves as expected with the changes as they are now, but would just like to know why this happens and if I can indeed minify the code a little as I described.